modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/TopLevelParameterTypeFactoryBean.java

```diff
@@ -13,6 +13,12 @@
 
 import org.springframework.beans.factory.FactoryBean;
 
+/**
+ * A {@link FactoryBean} that creates instances of a type defined by a
```

Comment:
Class name and javadoc do not match to me. (ie  class is "TopLevelParameterTypeFactoryBean" but there is no mention to top level and parameter in the javadoc)